pkg/reconciler/owner_references.go

```diff
@@ -17,30 +17,22 @@ limitations under the License.
 package reconciler
 
 import (
-	"fmt"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-
-	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
 )
 
-func kind(obj metav1.Object) schema.GroupVersionKind {
-	switch obj.(type) {
-	case *v1alpha1.Service:
-		return v1alpha1.SchemeGroupVersion.WithKind("Service")
-	case *v1alpha1.Route:
-		return v1alpha1.SchemeGroupVersion.WithKind("Route")
-	case *v1alpha1.Configuration:
-		return v1alpha1.SchemeGroupVersion.WithKind("Configuration")
-	case *v1alpha1.Revision:
-		return v1alpha1.SchemeGroupVersion.WithKind("Revision")
-	default:
-		panic(fmt.Sprintf("Unsupported object type %T", obj))
-	}
+// TODO(kubernetes/kubernetes#3030): Just use GroupVersionKind() from the
+// embedded metav1.TypeMeta
```

Comment:
FYI, this isn't always populated, so I'm not sure we can rely on this unless they are doing magic.